qemu/tests/data_plane_machine_type.py

```diff
@@ -0,0 +1,52 @@
+import logging
+
+from autotest.client.shared import error
+
+from virttest import utils_misc
+from virttest import env_process
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    boot with different machine type:
+    1) get supported machine type
+    2) boot guest with different machine types
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error.context("Get host's supported machine type", logging.info)
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    tmp = utils_misc.get_support_machine_type(qemu_binary)
+    (supported_machine_types, expected_system_versions) = tmp
+    q35_machine_list = []
+    i440fx_machine_list = []
+    
+    for m_type in supported_machine_types:
+        if "q35" in m_type:
+            q35_machine_list.append(m_type)
+        else:
+            i440fx_machine_list.append(m_type)
+
+    if params["machine_type"] == "pc":
+        machine_types = i440fx_machine_list
+    else:
+        machine_types = q35_machine_list
+  
+    error.context("Start vm with different machine_types")
+    for m_type in machine_types:
+        params["machine_type"] = m_type
+        params["start_vm"] = "yes"
+        vm_name = params['main_vm']
+        error.context("Boot the vm using -M option:'-M %s'"
+                      %m_type, logging.info)
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        if not vm.is_alive():
+            test.fail("Start vm with machine type:%s fail" %machine)
+        else:
+            error.context("Quit guest and check the process quit normally",
+                  logging.info)
+            vm.destroy(gracefully=False)    
```

Comment:
I'd suggest  `if  vm.is_alive(): ` test pass `else` test fail
Not sure if you need to do `vm.destroy(gracefully=False)` for each for loop, no matter it is fail or pass